backend/logic/users/users.js

```diff
@@ -67,7 +74,77 @@ var getCompletedCourses = async studentID => {
   return courses;
 };
 
+var login = async () => {
+	
+	let LocalStrategy = require('passport-local').Strategy;
+		
+	//LOCAL SIGNIN
+	app.passport.use('local-signin', new LocalStrategy(  
+	{
+		// by default, local strategy uses username and password, we will override with email
+		usernameField : 'username',
+		passwordField : 'password',
+		passReqToCallback : true // allows us to pass back the entire request to the callback
+	},
+	
+	function(req, username, password, done) {
+		let error;
+
+		let isValidPassword = function(userpass, password){
+			return bCrypt.compareSync(password, userpass);
+		}
+		// Check for missing input params
+		if (!username) {
+			error = 'Undefined username';
+		} else if (!password) {
+			error = 'Undefined password';
+		}
+
+		if (error) {
+			return done(error);
+		}
+
+		// Check for invalid formatting
+		if (!format.verifyUsername(username)) {
+			error = 'Invalid format username';
+		} else if (!format.verifyPassword(password)) {
+			error = 'Invalid format password';
+		}
+
+		if (error) {
+			return done(error);
+		}
+
+		// Begin transaction with database
+		try {
+			let connection = await mysql.getNewConnection()
+			await connection.beginTransaction();
+			let userInfo = await connection.query('SELECT * FROM users WHERE username = ?;',[username]);
+			await connection.commit();
+			connection.release();
+
+			if(!userInfo) {
+				return done(null, false, { message: 'Incorrect username.' });
+			}
+
+			if (!isValidPassword(userInfo[0].password, password)) {
```

Comment:
Since all data in the database is hashed using the crypto package and you just fetched all user information from the database, there is no need to compare with the bcrypt package

1) Hash input password with the hasher function in hash.js
2) Compare